workspace/filter-workspace-packages/src/index.ts

```diff
@@ -348,6 +355,20 @@ function matchPackagesByGlob<Pkg extends Package> (
   return Object.keys(graph).filter((parentDir) => micromatch.isMatch(parentDir, formattedFilter, { format }))
 }
 
+function matchPackagesByPrivacy<Pkg extends Package> (
+  graph: PackageGraph<Pkg>,
+  _private?: boolean
+): string[] {
+  if (_private === undefined) {
+    return []
+  }
+  const [privatePackages, publicPackages]: Array<PackageGraph<Pkg>> = partition(
+    ([, { package: { manifest } }]) => manifest.private === true,
+    Object.entries(graph)
+  ).map((packageGraphEntry) => Object.fromEntries(packageGraphEntry))
+  return Object.keys(_private ? privatePackages : publicPackages)
+}
+
```

Comment:
looks too complicated to me

```suggestion
function matchPackagesByPrivacy<Pkg extends Package> (
  graph: PackageGraph<Pkg>,
  _private?: boolean
): string[] {
  if (_private === undefined) {
    return []
  }
  const result: string[] = []
  for (const [key, { manifest } ] of Object.entries(graph)) {
    if (Boolean(manifest.private) === _private) {
      result.push(key)
    }
  }
  return key

```